machines/backupAndRestore/backupAndRestoreSetup.ts

```diff
@@ -57,46 +69,109 @@ export const backupAndRestoreSetupMachine = model.createMachine(
         on: {
           HANDLE_BACKUP_AND_RESTORE: {
             actions: [
-              'sendDataBackupAndRestoreSetupStartEvent',
               'setIsLoading',
               'unsetShouldTriggerAutoBackup',
+              'sendDataBackupAndRestoreSetupStartEvent',
             ],
-            target: 'checkSignIn',
+            target: 'fetchShowConfirmationInfo',
           },
         },
       },
+      fetchShowConfirmationInfo: {
+        /*TODO: Check if this call to store can be replaced by settings machine itself
+         *This would avoid extra read calls to storage
+         */
+        entry: 'fetchShowConfirmationInfo',
+        on: {
+          STORE_RESPONSE: [
+            {
+              cond: 'isConfirmationAlreadyShown',
+              actions: 'setShowConfirmation',
+              target: 'checkSignIn',
+            },
+            {
+              actions: ['unsetIsLoading', 'setShowConfirmation'],
+              target: 'selectCloudAccount',
+            },
+          ],
+        },
+        states: {},
+      },
+      selectCloudAccount: {
+        on: {
+          GO_BACK: {
+            actions: 'sendBackupAndRestoreSetupCancelEvent',
+            target: 'init',
+          },
+          PROCEED: [
+            {
+              actions: 'setIsLoading',
+              target: 'checkSignIn',
+            },
+          ],
+        },
+        states: {},
+      },
       checkSignIn: {
         invoke: {
           src: 'isUserSignedAlready',
           onDone: [
             {
               cond: 'isSignedIn',
-              actions: ['setProfileInfo', 'unsetIsLoading'],
+              actions: [
+                () => console.log('isSignedIn'),
+                'setProfileInfo',
+                'unsetIsLoading',
+              ],
               target: 'backupAndRestore',
             },
             {
               cond: 'isNetworkError',
-              actions: ['unsetIsLoading'],
+              actions: [() => console.log('isNetworkError'), 'unsetIsLoading'],
               target: 'noInternet',
             },
             {
-              actions: ['unsetIsLoading'],
-              target: 'selectCloudAccount',
+              cond: 'isNotSignedInIOSAndViaConfirmationFlow',
+              actions: [
+                () => console.log('isNotSignedInIOSAndViaConfirmationFlow'),
+                'unsetIsLoading',
+                'setErrorReasonAsAccountRequired',
+              ],
+              target: '.error',
+            },
+            {
+              cond: 'isNotSignedInIOS',
```

Comment:
Can this step ever be reached as one has to setup their iCloud accounts when they start using an iPhone?

Reply:
There can be scenario when the user logged out after app install or iCloud is disabled for Inji. At any of these instances we will only get `false`. So this condition can be hit multiple scenarios. These are some of the edge cases where this can happen

1. What if I try to setup without an apple account?
2. Deleting / removing the apple account?
4. turnoff iCloud access for Inji
5. turnoff iCloud itself?
6. Logging/signing out of apple account

Reply:
Valid points